### DISCUSSION TITLE
lbdb 0.49.1 does not include khard support

### DISCUSSION THREAD

**Discussion by arponero** [Homebrew/discussions#2394] [Casks]
Hi all,

After upgrading to 0.49.1, I just noticed that lbdbq does not search in my khard database anymore. The output shows the following:
/usr/local/bin/lbdbq: line 75: m_khard_query: command not found

And, actually, the m_khard module is not in lib/lbdb.

The original source does include the khard module, so I had to manually download the source, configure, make and make install it, and everything is working fine now.

Best,
Jaime

**Comment by SMillerDev** [comment]
I'm guessing this is about a formula instead of a cask since you managed to install it from source.

In which case, could you make a pull request to make the formula do that too?

**Reply by gwarf** [reply]
I'm also having this issue, and I tried to naively follow https://docs.brew.sh/FAQ#can-i-edit-formulae-myself, thus doing something like:
```shell
export HOMEBREW_NO_INSTALL_FROM_API=1
brew tap --force homebrew/core
brew edit lbdb
# adding `depends_on "khard"` after `depends_on "abook"`
# Assuming it should be enough thanks to the superenv magic...
brew reinstall lbdb
```

But it turns out that the module is not created/installed: 
```shell
ls /usr/local/Cellar/lbdb/0.54/lib/lbdb/
ABQuery                 m_bbdb                  m_muttalias             m_wanderlust
fetchaddr               m_fido                  m_osx_addressbook       munge
lbdb-munge              m_finger                m_palm                  palm_lsaddr
lbdb_bbdb_query.el      m_gnomecard             m_passwd                qpto8bit
lbdb_lib                m_inmail                m_pine                  tac
m_abook                 m_ldap                  m_vcf                   vcquery
```

Currently, I added the templated module from https://github.com/RolandRosenfeld/lbdb/blob/master/m_khard.sh.in as `~/.lbdb/modules/m_khard` and added it to the module path:
```shell
cat ~/.lbdbrc 
MODULES_PATH="MODULES_PATH $HOME/.lbdb/modules"

# https://github.com/RolandRosenfeld/lbdb/blob/master/lbdb.rc.in
METHODS="m_khard"
```
But it would be better to be able to rely on the brew bottle...

I would be happy to make a PR if I can get it work, but currently it's not the case, so I'm looking forward to any suggestion.

**Reply by SMillerDev** [reply]
You can submit your changes as a draft pull request and maybe I can help you get it fixed from there. It's much easier to debug with CI running.

**Reply by gwarf** [reply]
OK @SMillerDev, will do this, thanks!

**Reply by gwarf** [reply]
Here is the PR, thanks: https://github.com/Homebrew/homebrew-core/pull/181760.
Thanks to the PR template, apparently I was just missing a parameter:
```shell
brew reinstall --build-from-source lbdb
ls /usr/local/Cellar/lbdb/0.54/lib/lbdb/
ABQuery                 m_fido                  m_osx_addressbook       palm_lsaddr
fetchaddr               m_finger                m_palm                  qpto8bit
lbdb-munge              m_gnomecard             m_passwd                tac
lbdb_bbdb_query.el      m_inmail                m_pine                  vcquery
lbdb_lib                m_khard                 m_vcf
m_abook                 m_ldap                  m_wanderlust
m_bbdb                  m_muttalias             munge
```
Maybe `--build-fomr-source` should be specified in https://docs.brew.sh/FAQ#can-i-edit-formulae-myself? Or is it due to some other issue on my side?

**Reply by gromgit** [reply]
> Maybe `--build-fomr-souce` should be specified in https://docs.brew.sh/FAQ#can-i-edit-formulae-myself?

[It will be.](https://github.com/Homebrew/brew/pull/18098)

**Reply by gwarf** [reply]
Cool, thanks @gromgit!
